Remove some unrefenced primitives and update primitive table

Alisu opened the pull request:
As title says, see if CI pass.

Alisu committed:
Remove some unrefenced primitives and update primitive table

guillep commented on smalltalksrc/VMMaker/InterpreterPrimitives.class.st:
```diff
@@ -1821,36 +1821,6 @@ InterpreterPrimitives >> primitiveDivideLargeIntegers [
 ]
 
-{ #category : #'process primitives' }
-InterpreterPrimitives >> primitiveEnterCriticalSection [
```
These we should keep, and we should use in the image to have a proper mutex implementation

guillep commented on smalltalksrc/VMMaker/InterpreterPrimitives.class.st:
```diff
@@ -1885,31 +1855,6 @@ InterpreterPrimitives >> primitiveEqualLargeIntegers [
 
-{ #category : #'process primitives' }
-InterpreterPrimitives >> primitiveExitCriticalSection [
-	"Exit the critical section.
```
Same

guillep commented on smalltalksrc/VMMaker/CoInterpreterPrimitives.class.st:
```diff
@@ -246,108 +246,6 @@ CoInterpreterPrimitives >> primitiveCollectCogCodeConstituents [
 ]
 
-{ #category : #'indexing primitives' }
-CoInterpreterPrimitives >> primitiveContextXray [
```
I wouldn't remove this one yet, it could be useful in the future...

guillep commented on smalltalksrc/VMMaker/StackInterpreterMT.class.st:
```diff
@@ -1347,75 +1347,6 @@ StackInterpreterMT >> preemptionYields: aBoolean [
 ]
 
-{ #category : #'vm scheduling' }
-StackInterpreterMT >> primitiveEnterCriticalSection [
```
Same

guillep commented on smalltalksrc/VMMaker/StackInterpreterMT.class.st:
```diff
@@ -1347,75 +1347,6 @@ StackInterpreterMT >> preemptionYields: aBoolean [
-]
-
-{ #category : #'vm scheduling' }
-StackInterpreterMT >> primitiveExitCriticalSection [
```
Same

guillep commented on smalltalksrc/VMMaker/StackInterpreterPrimitives.class.st:
```diff
@@ -864,28 +864,6 @@ StackInterpreterPrimitives >> primitiveContextSize [
 ]
 
-{ #category : #'indexing primitives' }
-StackInterpreterPrimitives >> primitiveContextXray [
```
Same

Alisu committed:
Take into account PR comments, tough the primitiveTable stays unchanged

Alisu commented:
Btw i did not understand your message but i have a mutex implementation for the channel, i can push it if needed.

guillep commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -1386,14 +1386,14 @@ StackInterpreter class >> initializePrimitiveTable [
 		(180 primitiveGrowMemoryByAtLeast)
 		(181 primitiveSizeInBytesOfInstance)
-		(182 primitiveSizeInBytes)
+		(182 primitiveFail)			"primitiveSizeInBytes"
```
We should undo this line

guillep commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -1386,14 +1386,14 @@ StackInterpreter class >> initializePrimitiveTable [
-		(187 primitiveTestAndSetOwnershipOfCriticalSection) "was primitiveClosureValueWithArgs"
+		(185 primitiveFail)		"primitiveExitCriticalSection" "similar to signal hence index = signal + 100"
+		(186 primitiveFail) 			"primitiveEnterCriticalSection" "similar to wait hence index = wait + 100. was primitiveClosureValue"
+		(187 primitiveFail) "primitiveTestAndSetOwnershipOfCriticalSection""was primitiveClosureValueWithArgs"
```
We should undo these 3 lines

guillep commented on smalltalksrc/VMMaker/StackInterpreter.class.st:
```diff
@@ -1424,8 +1424,8 @@ StackInterpreter class >> initializePrimitiveTable [
-		(212 primitiveContextSize)
-		(213 primitiveContextXray)
+		(212 primitiveContextSize)	
+		(213 primitiveFail)			"primitiveContextXray"
```
We should undo this line

Alisu committed:
Update StackInterpreter.class.st

Follow requested changes, put back on primitive sizeInbytes, critical sections... and contextXray
